lib/wasix/src/journal/concrete/archived.rs

```diff
@@ -0,0 +1,3558 @@
+use lz4_flex::block::{compress_prepend_size, decompress_size_prepended};
+use num_enum::{IntoPrimitive, TryFromPrimitive};
+use rkyv::ser::{ScratchSpace, Serializer};
+use rkyv::{Archive, CheckBytes, Deserialize as RkyvDeserialize, Serialize as RkyvSerialize};
+use std::borrow::Cow;
+use std::time::SystemTime;
+use virtual_net::{Duration, IpAddr, IpCidr, Ipv4Addr, Ipv6Addr, SocketAddr, StreamSecurity};
```

Comment:
Can we use custom types for everthing in the journal?

Depending on types from separate contexts (virtual_net, SystemTime, wasmer_wasix_types) can make it easy to break the format accidentally, while we need this to be perma-stable.

As @syrusakbary mentioned, moving the journaling out to a separate crate (like `wasix_journal`) would help to ensure this.

Reply:
What I was mentioning was to have a `wasmer_journal` crate, independent from WASIX, and then WASIX can implement the types and such

Reply:
That doesn't make much sense to me.

This is a journal for WASIX syscalls, it's inherently tied to what syscalls WASIX has.

That doesn't mean it needs to be in the WASIX crate though.

Reply:
I guess you could argue that it's a syscalll journal, which happens to overlap with WASIX, but it doesn't really relate to Wasmer (the runtime).

Reply:
I would have to replicate all the WASIX enums to do this (and there are lots of them!).... including the infamous Errno enum - that's some serious boiler plate guys, we sure we really want to do this?

Reply:
Actually there might be a way to avoid that, let me give it a shot

Reply:
https://github.com/wasmerio/wasmer/pull/4263/commits/7c1b68df6b11c47c825ab2b01202a336c3e3370c****

Reply:
That commit moves out the crate however completely decoupling that crate from all other types will create a large amount of refactoring, needless boilerplate and duplication of code. Especially around enums and flags, etc...

@syrusakbary @theduke Are you really sure you want to do this? All that extra boilerplate looks like premature abstraction to me and really increases the code complexity while the actual on disk format (`ArchivedJournalEntry`) is already abstracted from those types and hence protected from changes that could impact it.

Reply:
I think the commit that @john-sharratt did is addressing what I commented perfectly. The PR looks good to me now

Reply:
I know it's a lot of boilerplate, *but*: we really need the journal to be perma-stable and to not have any breaking changes.

The types from `virtual-net` should definitely be duplicated, but those are not that many. 

I guess the WASIX enums and partially the types also more or less need to be perma-stable, but things will get more complicated if we add WASI preview 2 support, and if we make any kinds of changes to the representation of WASI types. 

My intuition is to go with all separate types.